WalletWasabi.Tests/UnitTests/Tor/Control/PipeWriterExtensions.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Tests.UnitTests.Tor.Control
+{
+	public static class PipeWriterExtensions
+	{
+		public static ValueTask<FlushResult> WriteAsync(this PipeWriter writer, string data, Encoding encoding, CancellationToken cancellationToken = default)
```

Comment:
I have encountered serious instability with Pipes here, it is questionable if the problem still occurs but definitely something to investigate before starting to use this. 
https://github.com/zkSNACKs/WalletWasabi/pull/4747#issuecomment-731263633

Reply:
I see your point. I looked into it and:

* https://github.com/dotnet/runtime/issues?q=is%3Aissue+label%3Aarea-System.IO.Pipelines+label%3ABug - meaning, there are no open bugs and it looks like this confirms what you said that Pipelines were buggy in .NET Core 3.x. 
* Piplines PRs: https://github.com/dotnet/runtime/pulls?page=1&q=is%3Apr+label%3Aarea-System.IO.Pipelines - I can't see any attempt to fix something really broken. Certainly when going through the PRs and issues it does not like Pipelines API is second class citizen or something.
* I have run `~/WalletWasabi/WalletWasabi.Tests$ rm a.log; for i in `seq 1000`; do echo "======= Run #$i =======" >>a.log; dotnet test --configuration Debug --filter "WalletWasabi.Tests.UnitTests.Tor.Control" --logger "console;verbosity=detailed" 2>&1 | tee -a a.log; done` on my Linux VM with .NET 5.0.5 and it did not fail.

Is that a sufficient proof for you?

Reply:
The problems were on macOS.

Reply:
Streams cannot be used here?

Reply:
> The problems were on macOS.

Can you describe the issue more in details? Would it make sense to report it?

> Streams cannot be used here?

I tried that way back but `MemoryStream` is [not thread-safe](https://stackoverflow.com/questions/12328245/memorystream-have-one-thread-write-to-it-and-another-read) so to make some tests possible one to implement it oneself. On the other hand, System.IO.Pipeslines fit our use case very well (I really recommend https://docs.microsoft.com/en-us/dotnet/standard/io/pipelines#what-problem-does-systemiopipelines-solve) as it shows all the possible mistakes one can make when writing networking stuff. So Pipelines make sense to me here.

Reply:
Also please note that in https://github.com/zkSNACKs/WalletWasabi/pull/4747#issuecomment-731263633 you used named pipes (https://docs.microsoft.com/en-us/dotnet/api/system.io.pipes.namedpipeserverstream?view=net-5.0) and I use https://docs.microsoft.com/en-us/dotnet/api/system.io.pipelines.pipe?view=dotnet-plat-ext-5.0 so it's not the same.